src/Microsoft.Diagnostics.DebugServices.Implementation/ManagedImageMappingModuleService.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.DebugServices.Implementation
+{
+    /// <summary>
+    /// Module service implementation for managed image mapping
+    /// </summary>
+    public class ManagedImageMappingModuleService : ModuleService
+    {
+        private readonly IRuntimeService _runtimeService;
+
+        public ManagedImageMappingModuleService(IServiceProvider services)
+            : base(services)
+        {
+            _runtimeService = services.GetService<IRuntimeService>();
+        }
+
+        /// <summary>
+        /// Get/create the modules dictionary.
+        /// </summary>
+        protected override Dictionary<ulong, IModule> GetModulesInner()
+        {
+            var modules = new Dictionary<ulong, IModule>();
+            int moduleIndex = 0;
+
+            IEnumerable<IRuntime> runtimes = _runtimeService.EnumerateRuntimes();
+            if (runtimes.Any())
+            {
+                foreach (IRuntime runtime in runtimes)
+                {
+                    ClrRuntime clrRuntime = runtime.Services.GetService<ClrRuntime>();
+                    if (clrRuntime is not null)
+                    {
+                        foreach (ClrModule clrModule in clrRuntime.EnumerateModules())
+                        {
+                            ModuleFromAddress module = new(this, moduleIndex, clrModule.ImageBase, clrModule.Size, clrModule.Name);
+                            module.ServiceContainer.AddService(clrModule);
```

Comment:
This should probably be a method in this case to avoid modifying containers from external entities. Also, I can't find a usage of `ClrModule` as a service, is this needed?

Reply:
The method would still allow some access to the container.  The ClrModule on the IModule services It isn't currently being used; it is there for some future thing I want to do.

Reply:
I removed adding ClrModule to the container for now and made ServiceContainer protected and renamed it to _s...